src/routing/approved.rs

```diff
@@ -56,6 +62,8 @@ pub(crate) struct Approved {
     msg_filter: MessageFilter,
     pub(super) event_tx: mpsc::UnboundedSender<Event>,
     joins_allowed: bool,
+    resource_proof: ResourceProof,
+    nonces: LruCache<XorName, [u8; 32]>,
```

Comment:
Can we avoid this LRU cache? How about we make it so the joining node needs to send the nonce back to us together with the solution? To avoid tampering, we could sign the nonce with the ed25519 key.

Reply:
> To avoid tampering, we could sign the nonce with the ed25519 key.

Yeah, good point. will change it.